mlflow/pyspark/ml/__init__.py

```diff
@@ -924,9 +938,44 @@ def _log_posttraining_metadata(estimator, spark_model, params):
 
         if log_models:
             if _should_log_model(spark_model):
-                # TODO: support model signature
+                from mlflow.models import infer_signature
+                from mlflow.pyspark.ml._autolog import (
+                    cast_spark_df_with_vector_to_array,
+                    get_feature_cols,
+                )
+                from mlflow.spark import find_and_set_features_col_as_vector_if_needed
+                from pyspark.sql import SparkSession
+
+                spark = SparkSession.builder.getOrCreate()
+
+                def _get_input_example_as_pd_df():
+                    feature_cols = list(get_feature_cols(input_df.schema, spark_model))
+                    limited_input_df = input_df.select(feature_cols).limit(
+                        INPUT_EXAMPLE_SAMPLE_ROWS
+                    )
+                    return cast_spark_df_with_vector_to_array(limited_input_df).toPandas()
+
+                def _infer_model_signature(input_example_slice):
+                    input_slice_df = find_and_set_features_col_as_vector_if_needed(
+                        spark.createDataFrame(input_example_slice), spark_model
+                    )
+                    model_output = spark_model.transform(input_slice_df)
```

Comment:
The `spark_model.transform` return a new dataframe which appends new columns (e.g. prediction column) to input dataframe 
i.e., the real output is only the "appended" columns (i.e. prediction column), but the `model_output` here contains both the original input columns and appended columns.

I think here we should extract these appended columns as the real output column. @dbczumar what do you think ?

Reply:
@WeichenXu123 Yes, I agree! @bali0019 can we make this change?